File: C++/freeCodeCamp.org_C++ Tutorial for Beginners/src/014_More If Statements/MoreIfStatements.cpp
#include <iostream>

using namespace std;

int getMax(int, int);
int getMax(int, int, int); // overrides the getMax() function if 3 parameters are parsed

int main()
{
    cout << getMax(2, 5) << endl;
    cout << getMax(2, 5, 3) << endl;

    return 0;
}

int getMax(int num1, int num2) // Compares two numbers and returns the bigger one
{
    int result;

    if(num1 > num2)
    {
        result = num1;
    } else {
        result = num2;
    }
    
    return result;
}

int getMax(int num1, int num2, int num3) // Does the same but with 3 numbers
{
    int result;

    if(num1 >= num2 && num2 >= num3)
    {
        result = num1;
    } else if(num2 >= num1 && num2 >= num3){
        result = num2;
    } else {
        result = num3;
    }
    
    return result;
}

File: Python/002_OOP Basics/014_More If Statements/MoreIfStatements.py
from functools import singledispatch

@singledispatch
def getMax(number):
    return number

@getMax.register
def _(number1, number2): # Compares two numbers and returns the bigger one
    if(number1 > number2):
        result = number1
    else:
        result = number2
    return result

@getMax.register
def _(number1, number2, number3): # Does the same but with 3 numbers
    if(number1 >= number2 and number2 >= number3):
        result = number1
    elif(number2 >= number1 and number2 >= number3):
        result = number2
    else:
        result = number3

print(getMax(2, 5))
print(getMax(2, 5, 3))